README.md

```diff
@@ -18,115 +18,230 @@ and used in apps configured by Kopytko Packager.
+`constructor()` and all of its other lifecycle methods, and initialize children Kopytko components.
+
+```brightscript
+' example of initializing Kopytko component in MainScene, assuming AppView extends KopytkoGroup
+app = m.top.createChild("AppView")
+app.id = "app"
+app.callFunc("initKopytko", {})
+```
+
+The `initKopytko` interface method has to be manually called also for components rendered via SceneGraph list and grid
+nodes (passed via `itemComponentName` property), usually in their `init` method (check the example
+in [Force update method](#force-update-method)).
 
 ### Constructor method
-This is similar to every class based language's constructor() method. This is where you'll define the initial value
+This is similar to every class based language's `constructor` method. This is where you'll define the initial value
 of the component state and any dependency it might need. It's called right after the component is created and all
-of its props are set, that means you can use m.top inside this method to access all the props that were passed
-to the component on its creation. You can see a simple example demonstrating the basic usage of the constructor() method
-in the image below.
+of its props are set, that means you can use `m.top` inside this method to access all the props that were passed
+to the component on its creation.
 
-![img.png](docs/constructor.png)
+A simple example demonstrating the basic usage of the `constructor` method:
+```brigthscript
+sub constructor()
+  m._arrayUtils = ArrayUtils()
 
-The main difference between constructor() and init() is that here you have access to the props that were set
-in the component, where in init() you don't since it's called right after the node is instantiated.
+  m.state = { focusedButtonIndex: 0 } ' setting initial state
+end sub
+```
+
+The main difference between `constructor()` and `init()` is that here you have access to the props that were set
+in the component, where in `init()` you don't since it's called right after the node is instantiated.
 
 ### Render method
-The render() method is the most important method in a Kopytko component and it must always be defined.
-It must return an object following an element model which must have at least a `name` property containing
-the name of the component to be rendered, additionally it can have a `props` property to define the fields
-and custom props of this element, an `events` property to define callbacks to the component field changes,
-and a `children` property which must be an array of this same object.
-With this single object you can create a whole tree of elements defining dynamic values where needed.
-You can see a simple example demonstrating the basic usage of the render() method in the image below.
+The `render` method is the most important method in a Kopytko component, and it must always be defined.
+It must return an object implementing the Element interface:
```

Comment:
It might return Invalid, Object or Array. Each of them has different implications. Can you describe them?